File: backend/phone.py
import re


def normalize_phone_e164(phone: str) -> str:
    raw = str(phone or "").strip()
    digits = re.sub(r"\D", "", raw)

    if len(digits) == 10:
        return f"+1{digits}"
    if len(digits) == 11 and digits.startswith("1"):
        return f"+{digits}"
    if raw.startswith("+") and 8 <= len(digits) <= 15 and not digits.startswith("0"):
        return f"+{digits}"
    raise ValueError("Phone number must be a valid E.164 or 10-digit US number")

File: src/phone.js
export function normalizePhone(phone) {
  const raw = String(phone || '').trim()
  const digits = raw.replace(/\D/g, '')

  if (digits.length === 10) return `+1${digits}`
  if (digits.length === 11 && digits.startsWith('1')) return `+${digits}`
  if (raw.startsWith('+') && digits.length >= 8 && digits.length <= 15 && digits[0] !== '0') {
    return `+${digits}`
  }
  return null
}
